### PR TITLE
Update and refine the Heroku deployment guide

### PR DESCRIPTION
**Pull request by shagedorn**
### Motivation:

Update to recent versions so newcomers can use examples from this guide directly without further changes. Additional context was added to make it easier to understand how Heroku works in general.

### Modifications:

- update to latest Heroku stack
- change default branch name to `main`
- fix outdated/broken links
- consistent formatting & capitalization
- additional links and high-level overview

### Result:

A hopefully easier-to-follow guide :)

### CODE REVIEW

File: server/guides/deploying/heroku.md [hunk 1]
```diff
@@ -3,99 +3,104 @@ layout: page
 title: Deploying to Heroku
 ---
 
-## What is Heroku
+Heroku is a popular all-in-one hosting solution.
 
-Heroku is a popular all in one hosting solution, you can find more at heroku.com
+For your Heroku application, you only need to select a virtual operating system ([stack](#stack-selection)) and preinstalled software packages ([buildpack](#buildback-selection)). After installing Heroku's CLI tools, you can use standard git commands to push your code to Heroku, where it will be built from source and deployed.
```
Comment (shagedorn) [inline server/guides/deploying/heroku.md:8]: I feel a high-level overview helps to decide between the various deployment options, especially for people (like me) with little server-side experience 🙂

File: server/guides/deploying/heroku.md [hunk 2]
```diff
@@ -3,99 +3,104 @@ layout: page
...
+## CLI Setup
 
-### HomeBrew
+Make sure that you've installed the Heroku CLI tools.
+
+### Homebrew Installation
 
 ```bash
-brew install heroku/brew/heroku
+brew tap heroku/brew && brew install heroku
```
Comment (shagedorn) [inline server/guides/deploying/heroku.md:23]: Copied directly from the [official guide](https://devcenter.heroku.com/articles/heroku-cli#install-the-heroku-cli).

File: server/guides/deploying/heroku.md [hunk 3]
```diff
@@ -3,99 +3,104 @@ layout: page
...
 
 ```bash
-$ heroku git:remote -a your-apps-name-here
+heroku git:remote -a your-apps-name-here
 ```
 
-### Set Stack
+### Stack Selection
 
-As of 13 September 2018, Heroku’s default stack is Heroku 18, we need it to be 16 for swift projects.
```
Comment (shagedorn) [inline server/guides/deploying/heroku.md:68]: I believe the last part is not needed anymore, `16` is in the end-of-life phase and unlikely to be useful for any new project.

File: server/guides/deploying/heroku.md [hunk 4]
```diff
@@ -3,99 +3,104 @@ layout: page
...
+brew tap heroku/brew && brew install heroku
 ```
 
 ### Other Install Options
 
-See alternative install options here: [https://devcenter.heroku.com/articles/heroku-cli#download-and-install](https://devcenter.heroku.com/articles/heroku-cli#download-and-install).
+See alternative install options [here](https://devcenter.heroku.com/articles/heroku-cli#install-the-heroku-cli).
 
-### Logging in
+### Login
```
Comment (0xTim) [inline server/guides/deploying/heroku.md:30]: Nit - should be two words
```suggestion
### Log In
```

File: server/guides/deploying/heroku.md [hunk 5]
```diff
@@ -106,14 +111,24 @@ You can use this command in terminal to set the file
...
+The contents of this file may vary depending on your server framework. For Vapor apps, the default Procfile content is the follwowing:
+
+```bash
+web: Run serve --env production --hostname 0.0.0.0 --port $PORT
+```
+
+### Wrap-Up
 
-We have now added the `.swift-version` file, and the `Procfile`, make sure these are committed into master or Heroku will not find them.
+Now that we have added the `.swift-version` and `Procfile` files, make sure these are committed on `main` to be included in your deployment.
```
Comment (0xTim) [inline server/guides/deploying/heroku.md:122]: I think for a tutorial like this, adding a couple of lines showing the `git add` and `git commit` commands is a good idea

File: server/guides/deploying/heroku.md [hunk 6]
```diff
@@ -106,14 +111,24 @@ You can use this command in terminal to set the file
...
 
-### Deploying to Heroku
+## Deployment to Heroku
 
-You're ready to deploy, run this from the terminal. It may take a while to build, this is normal.
+When you're ready to deploy, run this from the terminal:
 
-```none
+```bash
 git push heroku master
```
Comment (0xTim) [inline server/guides/deploying/heroku.md:129]: This should now be `main` right?

**Review by 0xTim — CHANGES_REQUESTED** [review]
Thanks for this! A few small changes but overall pretty much there!

File: server/guides/deploying/heroku.md (hunk 4 above)
Comment (shagedorn) [reply server/guides/deploying/heroku.md:30]: Wouldn't it be a single word, when written/interpreted as a noun? I tried to make it consistent to use primarily nouns for the section titles 🙂

File: server/guides/deploying/heroku.md (hunk 6 above)
Comment (shagedorn) [reply server/guides/deploying/heroku.md:129]: Absolutely, thanks! Will fix shortly.

**Comment by shagedorn** [comment]
I added a fixup commit for better change reviewability. Let me know if you want me to squash things together, or if you'll squash-merge via GitHub anyway (once you're happy with the changes).

**Comment by shagedorn** [comment]
@0xTim should be ready for another look, if you have a moment 🙂 🙏🏻

File: server/guides/deploying/heroku.md (hunk 4 above)
Comment (0xTim) [reply server/guides/deploying/heroku.md:30]: AFAIK "Login" isn't a real word - I'm just relaying what my editor told me circled in virtual giant red ink when I wrote a book 😆 but it's not a hill I'm overly attached to

**Review by 0xTim — APPROVED** [review]

**Comment by 0xTim** [comment]
@swift-ci please test

**Pull request merged**